framework/source/class/qx/io/request/AbstractRequest.js

```diff
@@ -433,7 +431,7 @@ qx.Class.define("qx.io.request.AbstractRequest",
     */
     sendWithPromise: function(context) {
       if (qx.core.Environment.get("qx.promise")) {
-        var context =  (context !== undefined && context !== null) ?
+        context = (context !== undefined && context !== null) ?
```

Comment:
More typically, this would be written as something like `context = context || this;` which is much simpler and easier to see what's going on. Is there a reason not to do that in this case?

Reply:
There is no particular reason for that. I was searching for what is the preferred way to set default function parameters in qooxdoo and it seemed more specific that way. 


I changed it and added two more tests for context binding. 


Another way would be using ES6 default function parameters so it would  be 
```javascript
sendWithPromise: function(context = this) {
```
but I have not seen it being used so far in framework's source so I was reluctant to use it.

Reply:
Since we still support the python compiler, which understands ECMA5 only, newer features are not yet allowed. So you did the right thing.